clients/src/main/java/org/apache/kafka/clients/admin/StreamsGroupDescription.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.clients.admin;
+
+import org.apache.kafka.common.GroupState;
+import org.apache.kafka.common.Node;
+import org.apache.kafka.common.acl.AclOperation;
+import org.apache.kafka.common.annotation.InterfaceStability;
+
+import java.util.Collection;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * A detailed description of a single streams group in the cluster.
+ */
+@InterfaceStability.Evolving
+public class StreamsGroupDescription {
+
+    private final String groupId;
+    private final int groupEpoch;
+    private final int targetAssignmentEpoch;
+    private final int topologyEpoch;
+    private final Collection<StreamsGroupSubtopologyDescription> subtopologies;
+    private final Collection<StreamsGroupMemberDescription> members;
+    private final GroupState groupState;
+    private final Node coordinator;
+    private final Set<AclOperation> authorizedOperations;
+
+    public StreamsGroupDescription(
+            final String groupId,
+            final int groupEpoch,
+            final int targetAssignmentEpoch,
+            final int topologyEpoch,
+            final Collection<StreamsGroupSubtopologyDescription> subtopologies,
+            final Collection<StreamsGroupMemberDescription> members,
+            final GroupState groupState,
+            final Node coordinator,
+            final Set<AclOperation> authorizedOperations
```

Comment:
Do we want to include the endpoint to partitions information in the description? I'm not sure.

Reply:
This can be derived - right? We know the endpoints for all members, their assingment, and the topology.